llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5696,6 +5710,10 @@ static void printTypeName(Type *ty, raw_ostream &nameStream) {
     nameStream << "]";
     return;
   }
+  if (auto pointerTy = dyn_cast<PointerType>(ty)) {
+    nameStream << "p" << pointerTy->getAddressSpace();
+    return;
+  }
```

Comment:
Suggest an assertion that `pointerTy` is either opaque or points to an i8.

Reply:
After adding this assertion I realized that all of this bitcasting may not be needed. Why do I think so? Because after adding this assertion I noticed that I missed one place and still everything was working (I missed adding changes to create row major matrix).
Now I'm thinking about revert bitcasts from this RP and change the way of creating functions.
Right now before creating new function we are checking if function (with the same name) is in the module. If function exists then we are using existing one, if not then we are creating new function.
I'm thinking about removing search of existing function and every time create a new function. From naming/type perspective there will not be any issue, because if function with specific name already exists then LLVM is adding suffix to the end of the newly created function (func, func.1, func.2). All of these functions can have the same argument types or different types.
I have to check (and you know you can save my time :) ) if there is a case in which we rely that only single function (with specific name and arguments) will be created and will have multiple users (multiple calls). The disadvantages of this solution is that we will end with functions which are having single users.
Thoughts ?

Reply:
Creating a new function each time may hurt the effectiveness of some optimizing transforms like CSE. I think the bitcasting approach is better, even if it's possible for some issue to slip through temporarily.

Reply:
@nhaehnle please check other version here: https://github.com/GPUOpen-Drivers/llpc/pull/2065